src/operator/sequence.ts

```diff
@@ -0,0 +1,99 @@
+// operator/sequence.ts
+
+/*
+ * Copyright (c) 2021 Check Digit, LLC
+ *
+ * This code is licensed under the MIT license (see LICENSE.txt for details).
+ */
+
+import type { Asyncerator } from '../asyncerator';
+
+import type { Operator } from './index';
+
+/**
+ * The sequenceFunction will be called repeatedly with an incrementing numerical parameter, returning a Promise
+ * that resolves with the same type as Input and is inserted into the stream.  The sequence operator
+ * passes through all other values.  Because the sequenceFunction returns a Promise, it
+ * can delay its response (using setTimeout) to emit values on a regular schedule, e.g. once a second.
+ *
+ * @param sequenceFunction
+ */
+export default function <Input>(sequenceFunction: (index: number) => Promise<Input>): Operator<Input, Input> {
+  return async function* (iterator: Asyncerator<Input>) {
+    const queue: Input[] = [];
+    let complete = false;
+    let hasThrown = false;
+    let errorThrown: unknown;
+
+    /**
+     * sequence producer
+     */
+
+    try {
+      (async () => {
+        // before we do anything, allow the event loop to process.  If the iterator completes immediately, we do not
+        // want any sequenceFunction execution.
+        await new Promise((resolve) => {
+          setTimeout(resolve, 0);
+        });
+
+        let currentIndex = 0;
+        // eslint-disable-next-line no-unmodified-loop-condition
+        while (!complete && !hasThrown) {
+          // eslint-disable-next-line no-await-in-loop
+          queue.push(await sequenceFunction(currentIndex++));
+
+          // sequenceFunction may resolve immediately, so we need to allow the event loop to process before repeating
+          // eslint-disable-next-line no-await-in-loop,no-loop-func
+          await new Promise((resolve) => {
+            setTimeout(resolve, 0);
+          });
+        }
+      })().catch((error: unknown) => {
+        hasThrown = true;
+        errorThrown = error;
+      });
+
+      /**
+       * pass-through producer
+       */
+
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      (async () => {
+        for await (const item of iterator) {
+          queue.push(item);
```

Comment:
It didn't need to before since I wasn't await-ing the producer promise.  But I do agree we should basically cancel this one when the sequence producer throws, so refactored a little.  (also in `race`)